eureka-client/src/main/java/com/netflix/discovery/InstanceInfoReplicator.java

```diff
@@ -0,0 +1,114 @@
+package com.netflix.discovery;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.discovery.util.RateLimiter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A task for updating and replicating the local instanceinfo to the remote server. Properties of this task are:
+ * - configured with a single update thread to guarantee sequential update to the remote server
```

Comment:
Good point. It should not be fatal in this case, as we still have the (guarded) heartbeat thread that will enforce re-registration for dirty instanceInfo (server side we check the lastDirtyTimestamp) if this thread is hanging on the network. This just means updates will be periodic as opposed to on-demand during the hangs until the hangs end.